sysutils/scp-backup/src/opnsense/scripts/OPNsense/ScpBackup/ScpBackup.php

```diff
@@ -53,13 +55,40 @@
 
     $remoteDirectoryFullPath = escapeshellarg($remoteDirectory . "config-" . date('Y-m-d-H-i') . ".xml");
 
-    $command = "scp -P $port -i $identifyFile $configFile $username@$hostname:$remoteDirectoryFullPath";
+    $tmpPrivateKeyFile = tempnam(sys_get_temp_dir(), TMP_PRIVATE_KEY);
+    $tmpHostKeyFile = tempnam(sys_get_temp_dir(), TMP_HOST_KEY);
+    $strictHostKeyParams = "-oStrictHostKeyChecking=no -oLogLevel=ERROR";
+
+    try {
+        if(file_put_contents($tmpPrivateKeyFile, $scpBackup->privatekey) !== false) {
 
-    syslog(LOG_WARNING, "scp_backup command: $command");
+            if($scpBackup->stricthostkeycheck == 1) {
+                $strictHostKeyParams = "-oStrictHostKeyChecking=yes";
+                if(!empty($scpBackup->hostkey) && file_put_contents($tmpHostKeyFile, $scpBackup->hostkey) !== false) {
+                    $strictHostKeyParams = $strictHostKeyParams . " -oUserKnownHostsFile=" . escapeshellarg($tmpHostKeyFile);
+                }
+            }
 
-    exec(escapeshellcmd($command), $output, $returnCode);
+            $command = "scp $strictHostKeyParams -P $port -i $tmpPrivateKeyFile $configFile $username@$hostname:$remoteDirectoryFullPath";
+            syslog(LOG_WARNING, "scp_backup command: $command");
+            exec(escapeshellcmd($command), $output, $returnCode);
```

Comment:
I am not sure if this gives unexpected behaviour due to double escaping

Reply:
Works correctly for me after testing.

Reply:
The PHP documentation says that this is for all arguments: https://secure.php.net/manual/en/function.escapeshellcmd.php - however, the strings that need escaping, are already escaped: L46, 47, 56, 68. Port is validated to be an integer so it cannot be used for command injection